jina/helper.py

```diff
@@ -909,6 +909,48 @@ def __delete__(self, obj):
             del obj.__dict__[f'CACHED_{self.func.__name__}']
 
 
+class _cache_invalidate(object):
+    """Class for cache invalidation, remove strategy.
+
+    :param func: func to wrap as a decorator.
+    :param func_to_invalidate: String as the function name to invalidate cached
+        data. E.g. in :class:`cached_property` we cache data inside the class obj
+        with the `key`: `CACHED_{func.__name__}`, the func name in `cached_property`
+        is the name to invalidate.
+    """
+
+    def __init__(self, func, func_to_invalidate: str):
+        self.func = func
+        self.func_to_invalidate = func_to_invalidate
+
+    def __call__(self, *args, **kwargs):
+        obj = args[0]
+        cached_key = f'CACHED_{self.func_to_invalidate}'
```

Comment:
what is the impact of having this extra `CACHED` that would add an extra entry to the dict. I think this can be removed from cached_property as well as it was used previously in the `executor` dump and is not used anymore

Reply:
@JoanFM i didn't get your idea,  how to we cache if we remove `CACHE` from `cached_property`?

Reply:
do we need to add the CACHED prefix? why do not we use the original attribute?

Reply:
it is a design already exist in `class:cached_property`, so I did kept it consistent here

Reply:
I know, I think is not needed there either

Reply:
okay clear @hanxiao shall we remove the `CACHED` prefix here, can you give some feedback?

Reply:
You need some prefix, otherwise you might overwrite the function definition itself. you can try removing the prefix, but I believe it will not work. Can you test it?

Reply:
i did remove the `CACHED_` prefix and runed all primitive types unit test, it worked actually